service.rb

```diff
@@ -43,10 +43,22 @@ def initialize(port)
           line = client.readline
           if line == "\r\n"
             body = client.read(content_length)
+            break
           end
         end
 
-        response_message =  "WIP: POST RESPONSE"
+        new_daily_data = Hash[URI.decode_www_form(body)]
```

Comment:
You should `require 'uri'` before you refer to the `URI` constant, because it comes from the [standard library](https://rubydoc.info/stdlib) which isn’t loaded by default.

I don’t know the reason but it looks like this works differently on the system Ruby vs one that you’ve installed yourself, which is perhaps why you didn’t notice a problem:

```
% chruby system
% ruby --version
ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
% ruby -e 'URI'
% chruby 2.6
% ruby --version
ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin20]
% ruby -e 'URI'
Traceback (most recent call last):
-e:1:in `<main>': uninitialized constant URI (NameError)
```

Reply:
Mine is slightly different where the `2.6` version is at a `darwin19` instead of a `darvin20` like yours? 

```
➜ chruby system
➜ ruby --version
ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
➜ ruby -e "URI"
➜ chruby 2.6
➜ ruby --version
ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
➜ ruby -e "URI"
```

Anyway, will add the require so it is environment-proof.